plugins/code-coverage/src/components/CoverageHistoryChart/CoverageHistoryChart.tsx

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import {
+  Box,
+  Card,
+  CardContent,
+  MenuItem,
+  Select,
+  Typography,
+} from '@material-ui/core';
+import {
+  LineChart,
+  XAxis,
+  YAxis,
+  Tooltip,
+  Legend,
+  Line,
+  CartesianGrid,
+  ResponsiveContainer,
+} from 'recharts';
+import { useAsync } from 'react-use';
+import { useApi } from '@backstage/core-api';
+import { useEntity } from '@backstage/plugin-catalog-react';
+import { codeCoverageApiRef } from '../../api';
+import { Progress } from '@backstage/core';
+import { Alert } from '@material-ui/lab';
+import TrendingDownIcon from '@material-ui/icons/TrendingDown';
+import TrendingUpIcon from '@material-ui/icons/TrendingUp';
+import TrendingFlatIcon from '@material-ui/icons/TrendingFlat';
+
+type Coverage = 'line' | 'branch';
+
+const getTrendIcon = (trend: number) => {
+  switch (true) {
+    case trend > 0:
+      return <TrendingUpIcon style={{ color: 'green' }} />;
+    case trend < 0:
+      return <TrendingDownIcon style={{ color: 'red' }} />;
+    case trend === 0:
+    default:
+      return <TrendingFlatIcon />;
+  }
+};
+
+export const CoverageHistoryChart = () => {
+  const { entity } = useEntity();
+  const codeCoverageApi = useApi(codeCoverageApiRef);
+  const {
+    loading: loadingHistory,
+    error: errorHistory,
+    value: valueHistory,
+  } = useAsync(
+    async () =>
+      await codeCoverageApi.getCoverageHistoryForEntity({
+        kind: entity.kind,
+        namespace: entity.metadata.namespace || 'default',
+        name: entity.metadata.name,
+      }),
+  );
+
+  if (loadingHistory) {
+    return <Progress />;
+  } else if (errorHistory) {
+    return <Alert severity="error">{errorHistory.message}</Alert>;
+  }
+
+  if (!valueHistory.history.length) {
+    return (
+      <Box>
+        <Card>
+          <CardContent>
+            <Box mb={2} display="flex" justifyContent="space-between">
+              <Typography variant="h5">History</Typography>
+            </Box>
```

Comment:
Probably want to use `<CardHeader>` [from MUI](https://material-ui.com/components/cards/) for these.